protocol-units/bridge/chains/movement/src/event_monitoring.rs

```diff
@@ -0,0 +1,177 @@
+use crate::MovementClient;
+use crate::{
+	event_types::{CounterpartyEventKind, MovementChainEvent},
+	types::MovementHash,
+	utils::MovementAddress,
+};
+use anyhow::Error;
+use anyhow::Result;
+use aptos_sdk::rest_client::Response;
+use aptos_types::contract_event::EventWithVersion;
+use async_stream::try_stream;
+use bridge_shared::bridge_monitoring::{
+	BridgeContractCounterpartyEvent, BridgeContractCounterpartyMonitoring,
+};
+use bridge_shared::types::{CounterpartyCompletedDetails, LockDetails};
+use futures::{Stream, TryFuture};
+use std::{pin::Pin, task::Poll};
+use tokio::sync::mpsc::UnboundedReceiver;
+
+#[allow(unused)]
+pub struct MovementCounterpartyMonitoring<A, H> {
+	listener: UnboundedReceiver<MovementChainEvent<A, H>>,
+	client: Option<MovementClient>,
+}
+
+impl BridgeContractCounterpartyMonitoring
+	for MovementCounterpartyMonitoring<MovementAddress, MovementHash>
+{
+	type Address = MovementAddress;
+	type Hash = MovementHash;
+}
+
+#[allow(unused)]
+impl MovementCounterpartyMonitoring<MovementAddress, MovementHash> {
+	async fn run(
+		_rest_url: &str,
+		_listener: UnboundedReceiver<MovementChainEvent<MovementAddress, MovementHash>>,
+	) -> Result<Self, anyhow::Error> {
+		todo!()
```

Comment:
Perhaps it would make sense to list the todos in the Outstanding Issues section of the PR?